kafka/adminapi.go

```diff
@@ -258,6 +258,14 @@ type ListConsumerGroupsResult struct {
 	Errors []error
 }
 
+// DeleteRecordsResult represents the result of a DeleteRecords call.
+type DeleteRecordsResult struct {
+	// A slice of TopicPartitions, with the offset field set to the post-deletion
+	// low-watermark offset (smallest available offset of all live replicas).
+	// The error is set if any occurred for that topic partition.
+	TopicPartitions []TopicPartition
+}
```

Comment:
Better to add a DeletedRecords struct like in Java and Python. That'll allow to be open to changes adding fields to it, like for example the number of deleted records. That is possible with .NET too by adding a DeletedRecords field to DeleteRecordsResult.

What's not possible without changing .NET API too is to set a RecordsToDelete object, but given it's not possible to delete records in different ways I don't see possibility of extension on that front.

I suggest this:

```suggestion
// DeletedRecords contains information about deleted
// records of a single partition
type DeletedRecords struct {
	// Low-watermark offset after deletion
	LowWatermark Offset
}

// DeleteRecordsResult represents the result of a DeleteRecords call
// for a single partition.
type DeleteRecordsResult struct {
	// One of requested partitions.
	// The Error field is set if any occurred for that partition.
	TopicPartition TopicPartition
	// Deleted records information, or nil if an error occurred.
	DeletedRecords *DeletedRecords
}

// DeleteRecordsResults represents the results of a DeleteRecords call.
type DeleteRecordsResults struct {
	// A slice of DeleteRecordsResult, one for each requested topic partition.
	DeleteRecordsResults []DeleteRecordsResult
}
```